Src/CSharpier/SyntaxPrinter/OptionalBraces.cs

```diff
@@ -0,0 +1,22 @@
+using System.Collections.Generic;
+using System.Linq;
+using CSharpier.DocTypes;
+using CSharpier.SyntaxPrinter.SyntaxNodePrinters;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace CSharpier.SyntaxPrinter
+{
+    public static class OptionalBraces
+    {
+        public static Doc Print(StatementSyntax node, string? groupId = null)
```

Comment:
I may be drinking the pattern matching cool-aid too much, but do you think this is more readable?
```cs
public static Doc Print(StatementSyntax node, string? groupId = null) =>
    (node, groupId) switch
    {
        (BlockSyntax blockSyntax, null) => Block.Print(blockSyntax),
        (BlockSyntax blockSyntax, var id) => Block.PrintWithConditionalSpace(
            blockSyntax,
            id
        ),
        (var otherSyntax, _) => Doc.Indent(Doc.HardLine, Node.Print(otherSyntax))
    };
```

Though, I do admit my ability to follow nested ternary operator blocks is somewhat subpar.

Reply:
That is less readable to me, but it is also new to me.
I was also not super happy with the nested ternary, but it seemed "good enough"